### PR TITLE
PWN-6129 - make ScanQrFragment changes

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6129

## Description of Work

PWN-6129 - make ScanQrFragment.kt small validation, replace fragment except of add, fixes of UiKitSnackbar.kt to apply onDismissed action even if you have actionButtonListener installed, apply search result with FragmentResultListener

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt [hunk 1]
```diff
@@ -52,10 +72,26 @@ class ScanQrFragment :
...
+                        onDismissed = continueAction,
+                        actionBlock = { continueAction.invoke() }
+                    )
+                }
+            } ?: showUiKitSnackBar(
+                messageResId = R.string.qr_common_error,
+                actionButtonResId = android.R.string.ok,
+                onDismissed = continueAction,
+                actionBlock = { continueAction.invoke() }
+            )
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt:94]: hard to read. 
maybe you should check `rawResult` for null first?

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt:94]: hmm ok i will extract it to methods

File: app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt [hunk 2]
```diff
@@ -52,10 +72,26 @@ class ScanQrFragment :
...
     private val barcodeCallback: ZXingScannerView.ResultHandler =
         ZXingScannerView.ResultHandler { rawResult ->
-            rawResult?.text?.let {
-                successCallback?.invoke(it)
-                popBackStack()
-            }
+            val continueAction: () -> Unit = { startCameraPreview() }
+            rawResult?.text?.let { address ->
+                try {
+                    PublicKey(address)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt:78]: PublicKeyValidator.isValid(address)

File: app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt [hunk 3]
```diff
@@ -52,10 +72,26 @@ class ScanQrFragment :
...
+                    setFragmentResult(requestKey, bundleOf(resultKey to address))
+                    popBackStack()
+                } catch (e: Throwable) {
+                    showUiKitSnackBar(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/qr/ui/ScanQrFragment.kt:82]: Timber?

File: app/src/main/res/layout/fragment_scan_qr.xml [hunk 4]
```diff
@@ -21,49 +21,71 @@
...
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="ContentDescription" />
 
-    <TextView
-        android:id="@+id/cameraPermissionRequestButton"
+    <LinearLayout
+        android:id="@+id/layoutCameraPermission"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_scan_qr.xml:48]: containerCameraPermisson

File: ui-kit/src/main/kotlin/org/p2p/uikit/natives/UiKitSnackbar.kt [hunk 5]
```diff
@@ -40,7 +41,7 @@ fun View.showSnackbarShort(
         buttonAction = actionButtonListener,
         duration = Snackbar.LENGTH_SHORT,
         style = style
-    )
+    ).addOnDismissedCallback(onDismissed)
         .show()
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/natives/UiKitSnackbar.kt:45]:
```suggestion
    )
        .addOnDismissedCallback(onDismissed)
        .show()
```

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**